src/tests/ftest/container/fill_destroy_loop.py

```diff
@@ -25,42 +24,43 @@ class BoundaryPoolContainerSpace(TestWithServers):
+        free_space = self.pool.get_pool_free_space()
+        self.log.info("--pool free space before writing data to container= %s", free_space)
+
+        # Write random data to container until pool out of space
+        base_data_size = container.data_size.value
+        data_written = 0
+        while True:
+            new_data_size = random.randint(base_data_size * 0.5, base_data_size * 1.5)  # nosec
+            container.data_size.update(new_data_size, "data_size")
+
             try:
-                written_byte += data_size
-                self.d_log.debug("--{0} writing cont-obj {1}  byte data, total {2}..".format(
-                    write_count, data_size, written_byte))
-                container.written_data.append(TestContainerData(False))
-                container.written_data[-1].write_record(
-                    container, akey, dkey, data_str, obj_class=cont_obj_class)
-                self.d_log.debug("--{0}wrote cont-obj, sz {1}".format(write_count, data_size))
-                write_count += 1
-            except DaosTestError as excep:
-                space_avail_to_write = False
-                if self.DER_NOSPACE in repr(excep):
-                    self.log.info("--(3)written_byte= %s, Der_no_space %s detected, pool is "
-                                  "unable for an additional %s byte object", written_byte,
-                                  self.DER_NOSPACE, data_size)
-                    free_space = self.pool.get_pool_free_space()
-                    self.log.info("--(4)free_space when pool is fulfill= %s", free_space)
-                else:
-                    self.fail("#Exception while writing object: {}".format(repr(excep)))
+                container.write_objects()
+            except TestFail as excep:
+                # Uncomment following for debugging
+                # self.log.info("%s", repr(excep))
+                if self.DER_NOSPACE in str(excep):
+                    self.log.info(
+                        "--(3) Der_no_space %s detected, pool is unable for an additional"
+                        " %s byte object", self.DER_NOSPACE, container.data_size.value)
+                    break
+                self.fail("Exception while writing object: {}".format(repr(excep)))
+            data_written += new_data_size
+
+        # display free space and data written
+        free_space_before = self.pool.get_pool_free_space()
+        self.log.info("--(4)%s bytes written when pool is full.", data_written)
+
+        # destroy container and check for free space increase
         container.destroy()
+        free_space_after = self.pool.get_pool_free_space()
+        self.log.info("--(5)pooll full, free_space before container delete= %s", free_space_before)
+        self.log.info("--(6)pooll full, free_space after  container deleted= %s", free_space_after)
+        if free_space_after <= free_space_before:
+            self.fail("Deleting container did not free up pool space.")
```

Comment:
Do we need to wait for aggregation? I'm not sure if free space is immediately returned